homeassistant/helpers/area_registry.py

```diff
@@ -29,26 +28,25 @@ class EventAreaRegistryUpdatedData(TypedDict):
     area_id: str
 
 
-@attr.s(slots=True, frozen=True)
+@dataclasses.dataclass(frozen=True, kw_only=True, slots=True)
 class AreaEntry:
     """Area Registry Entry."""
 
-    name: str = attr.ib()
-    normalized_name: str = attr.ib()
-    aliases: set[str] = attr.ib(
-        converter=attr.converters.default_if_none(factory=set)  # type: ignore[misc]
-    )
-    id: str | None = attr.ib(default=None)
-    picture: str | None = attr.ib(default=None)
-
-    def generate_id(self, existing_ids: Container[str]) -> None:
-        """Initialize ID."""
-        suggestion = suggestion_base = slugify(self.name)
+    aliases: set[str]
+    id: str | None
```

Comment:
Is `id=None` a valid use case or is it a leftover from the previous `generate_id` functionality?

Reply:
this is indeed not valid, thanks @edenhaus 👍